### PR TITLE
port pymol: update to 2.4.0, introduce x11 variant and use mmtf-cpp

### PR DESCRIPTION
**Pull request by jwhowarth**
port pymol: This commit updates the pymol package to the 2.4.0 release, adds upstream patches for the deprecated string.split python call, introduces an x11 variant and uses the default mmtf-cpp support . The APBS Tools plugin requires pmg_tk (X11) in order to run. The pdb2pqr functionality of APBS Tools also requires python2.7. So the x11 variant defaults to python27 while the -x11 variant defaults to python38. The x11 variant switches the default options.gui to pmg_tk to prevent pymol from favoring the pmg_qt gui.

###### Type(s)

- [ ] bugfix
- [x ] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.4 19E287
Xcode 11.5 11E608c 

###### Verification 
Have you

- [ x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [ x] checked your Portfile with `port lint`?
- [ x] tried existing tests with `sudo port test`?
- [ x] tried a full install with `sudo port -vst install`?
- [x ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by jwhowarth** [comment]
This newest permutation of the pymol packaging will require the new mmtf-cpp package to be added to macports.macports-ports.

https://github.com/macports/macports-ports/pull/7199

File: science/pymol/Portfile [hunk 1]
```diff
@@ -20,58 +20,77 @@ platforms           darwin
...
+}
+
+variant x11 {
+    depends_lib-append port:freeglut \
+                       port:mesa \
+                       port:py${python.version}-pmw \
+                       port:py${python.version}-tkinter \
+                       port:py${python.version}-tkinter \
+                       port:tcl \
+                       port:tk
```
Comment (reneeotten) [inline science/pymol/Portfile:78]: these whitespaces are not multiples of four

**Comment by jwhowarth** [comment]
The requested changes are in a new pull request at...

https://github.com/macports/macports-ports/pull/7202

The mechanism that I am using for GitHub Desktop to push the commit into my jwhowarth/macports-ports seems to block me from putting the new pull requests on its own branch.

**Comment by reneeotten** [comment]
I am not sure, I always do it on the command-line.... One thing I noticed earlier is that you seem to push to the `master` branch on your fork, that might be one of the issues. Ideally, you should make a new branch and submit a PR from there. Or do a bit of reading (for example [WorkingWithGit](https://trac.macports.org/wiki/WorkingWithGit) and try the command line next time ;) Anyway, it's almost good to go so it's fine to open/close one more PR!

**Pull request closed without merging**